Make required metadata fields more visible

Arnei opened the pull request:
Helps with #1565.

Overall this PR aims at making the user more aware of required fields in the metadata tab, particularly if one of those required fields is missing a value which will result in an error in the backend. The individual changes can be found in the commit history.

Does not prevent the user from clicking "Save & Process" even if there are still validation issues. Would like to add that in a different PR as a more generalized version of #1517.

### How to test this

Ideally this is tested with an Opencast you can control, so you can change which metadata fields are required in `etc/org.opencastproject.ui.metadata.CatalogUIAdapterFactory-episode-common.cfg`.

Arnei committed:
Fix validation for dropdowns

Metadata validation for dropdowns was no working,
this should help.

Arnei committed:
Fix metadata validation not on initial load

Validation for the metadata form was ignored for the
initial metadata values. This fixes that.

Arnei committed:
Mark required fields

Mark required metadata fields with a "*"
like in the admin ui. This does not change
any functionality.

Arnei committed:
Allow for multiple validators

Instead of returning only one validation function per field,
allow returning multiple functions . For example, "duration"
may be required but should also be formatted correctly.

Arnei committed:
Merge branch 'main' into improve-metadata-validation

wsmirnow commented:
`mixed_text` metadata field type seems not to work correctly. On page load missing values are accepted (see screenshot). But if you put some text there and delete it afterwards, the required-message is shown.

![image](https://github.com/user-attachments/assets/b31dddd9-97bb-4689-bd5f-4c47c389f9a7)

Arnei committed:
Fix metadata validation for arrays

Some metadata fields, like fields of type "mixed_text",
 store their value as arrays. Our `required` check did not
 handle arrays, but it should now.

wsmirnow commented on src/main/Metadata.tsx:
```diff
@@ -278,7 +284,19 @@ const Metadata: React.FC = () => {
+      val = value.submitValue;
+    }
+
+    if (value && Array.isArray(value) && value.length === 0) {
+      val = false;
+    }
```
In case you fill out `mixed_text` metadata value and delete the content afterwords, the required message isn't shown as expected. The value is `['']` in this case. Lets get rid of empty strings from array before checking the length.

```suggestion
    if (value && Array.isArray(value) && value.filter(v => v).length === 0) {
      val = false;
    }
```

Arnei replied:
Can't reproduce this.

[Bildschirmaufzeichnung vom 2025-05-28 10-08-07.webm](https://github.com/user-attachments/assets/96a7104a-0de9-48fc-925e-186a17c0c7ff)

Could you tell me what I'm doing wrong, or have any other tips for me on how to reproduce this? I worry that empty strings in the array point at a more fundamental problem.

wsmirnow replied:
Sure, here is an example

[editor-required-metadata-field-on-focus-lost.webm](https://github.com/user-attachments/assets/3cd6ef33-02fe-4bad-b8ba-568d9d520a50)

The affected metadata field configuration looks like

![image](https://github.com/user-attachments/assets/fd7a343a-3183-4894-bdaf-f49229248d27)

wsmirnow replied:
The delimiter configuration is not applying too. But this is a separate issue.

Arnei replied:
Thanks for the detailed report, I forgot to consider that `mixed_text` fields may not have a listprovider defined. The latest commit 74c72b should fix the issue you were seeing.

I agree that respecting delimiter configuration is better handled in a separate issue/PR.

wsmirnow replied:
Yes, thanks.

Arnei committed:
Fix empty mixed_text metadata fields not being validated correctly if they were empty

Mixed text fields usually have an array as their value
but have to be rendered as strings if no listprovider is
defined for them.
This improves on parsing the values for mixed text fields
if their are strings, to avoid having "['']" as their value.

gregorydlogan commented:
@wsmirnow are you concerns resolved?
